openapi3filter/req_resp_decoder.go

```diff
@@ -1221,3 +1224,44 @@ func FileBodyDecoder(body io.Reader, header http.Header, schema *openapi3.Schema
 	}
 	return string(data), nil
 }
+
+// ZipFileBodyDecoder is a body decoder that decodes a zip file body to a string.
+func ZipFileBodyDecoder(body io.Reader, header http.Header, schema *openapi3.SchemaRef, encFn EncodingFn) (interface{}, error) {
+	buff := bytes.NewBuffer([]byte{})
+	size, err := io.Copy(buff, body)
+	if err != nil {
+		return nil, err
+	}
+
+	zr, err := zip.NewReader(bytes.NewReader(buff.Bytes()), size)
```

Comment:
Why copy? Isn't it possible to read from `body` directly?

Reply:
Yes, it is not possible to read directly from `body`.
It must be converted to `io.ReaderAt`.
https://pkg.go.dev/archive/zip#NewReader